spec/lang/types.md

```diff
@@ -178,6 +185,19 @@ impl Type {
 And we also define how to compute the actual size and alignment.
 
 ```rust
+impl TupleHeadLayout {
+    // This is one function to avoid quadratic computations.
+    pub fn compute_size_align(self, tail_size: Size, tail_align: Align) -> (Size, Align) {
+        // TODO: support packed self
+        let align = tail_align.max(self.min_align);
```

Comment:
Isn't it a bug if tail_align < min_align?

Reply:
Why? A struct can contain 8 aligned fields, but the tail might only need to be 1 aligned.

Reply:
Ah of course.